File: 3423. Maximum Difference Between Adjacent Elements in a Circular Array/maxAdjacentDistance.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAdjacentDistance = function (nums) {
    let ans = 0;
    let len = nums.length

    for (let i = 1; i < len + 2; i++) {
        let oper = Math.abs(nums[i % len] - nums[(i - 1) % len])
        ans = Math.max(ans, oper)
    }

    return ans;
};

File: 3423. Maximum Difference Between Adjacent Elements in a Circular Array/maxAdjacentDistance.cpp
#include <iostream>
#include <vector>
#include <math.h>

using namespace std;

class Solution {
    public:
        int maxAdjacentDistance(vector<int>& nums) {
            int ans = 0;
            int len = nums.size();
    
            for(int i=1; i<len+2; i++){
                ans = max(ans, abs(nums[i%len] - nums[(i-1)%len]));
            }
    
            return ans;
        }
    };